libraries/provider_add_to_list.rb

```diff
@@ -48,17 +48,25 @@ def action_edit
             f.each_line do |line|
               if line =~ regex
                 found = true
-                if new_resource.delim.count == 1
-                  unless line =~ /(#{new_resource.delim[0]}|#{new_resource.pattern})\s*#{new_resource.entry}\s*(#{new_resource.delim[0]}|\n)/
-                    line = line.chomp + "#{new_resource.delim[0]}#{new_resource.entry}"
+                if new_resource.ends_with
+                  unless line =~ /(#{new_resource.delim[0]}|#{new_resource.pattern}).+#{new_resource.entry}(#{new_resource.delim[0]}|#{new_resource.ends_with}|\n)/
+                    list_end = line.rindex(new_resource.ends_with)
+                    line = line.chomp.insert(list_end, "#{new_resource.delim[0]}#{new_resource.entry}")
```

Comment:
list_end is nil if the line doesn't match new_resource.ends_with. I think the regex should check for ends_with being present.  In cases where ends_with includes ) or ] the values also need to be escaped.